### PR TITLE
[AC-358] Server changes for self host subscription page changes 

### PR DESCRIPTION
**Pull request by shane-melton**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Add a separate `SubscriptionExpiration` field to the organization license so that the grace period and subscription expiration can be differentiated. Additionally, add a new field to the self hosted organization subscription response that includes the new `SubscriptionExpiration` field from the license.

Associated clients PR can found here: https://github.com/bitwarden/clients/pull/5101

## Code changes

* **src/Core/Models/Business/OrganizationLicense.cs** Add new `SubscriptionExpiration` field and increment the `CURRENT_LICENSE_FILE_VERSION`.

* **src/Api/Models/Response/Organizations/OrganizationResponseModel.cs:** Update constructor logic to support accepting an organization license and setting the expiration dates accordingly.

* **src/Api/Controllers/OrganizationsController.cs:** Use the license service to read a self hosted license file when retrieving the self hosted organization's subscription information; use it to populate the expiration dates used by the client.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Core/Models/Business/OrganizationLicense.cs [hunk 1]
```diff
@@ -133,10 +135,10 @@ public OrganizationLicense()
     /// <summary>
     /// Represents the current version of the license format. Should be updated whenever new fields are added.
     /// </summary>
-    private const int CURRENT_LICENSE_FILE_VERSION = 10;
+    private const int CURRENT_LICENSE_FILE_VERSION = 12;
```
Comment (r-tome) [inline src/Core/Models/Business/OrganizationLicense.cs:138]: This constant is intentionally set to be a version behind. https://bitwarden.slack.com/archives/C02LPDKPML6/p1668134749839749

**Review by r-tome — COMMENTED** [review]
It all looks great to me besides needing to set `CURRENT_LICENSE_FILE_VERSION = 11`.

File: src/Core/Models/Business/OrganizationLicense.cs (hunk 1 above)
Comment (shane-melton) [reply src/Core/Models/Business/OrganizationLicense.cs:138]: Ah, I should have realized it was behind by one for some reason, thanks for catching that!

Comment (shane-melton) [reply src/Core/Models/Business/OrganizationLicense.cs]: Set `CURRENT_LICENSE_FILE_VERSION=10` and added comment explaining why its one behind the actual current version to hopefully catch this for anyone else in the future.

**Review by r-tome — DISMISSED** [review]
Excellent work!

**Review by r-tome — APPROVED** [review]

**Review by r-tome — APPROVED** [review]

**Review by r-tome — APPROVED** [review]

**Pull request merged**